File: PeriodisationProgramApp.Domain/Entities/ExerciseMuscleGroup.cs
using PeriodisationProgramApp.Domain.Enums;

namespace PeriodisationProgramApp.Domain.Entities
{
    public class ExerciseMuscleGroup : BaseEntity
    {
        public Guid ExerciseId { get; set; }

        public MuscleGroup? MuscleGroup { get; set; }

        public MuscleGroupRole MuscleGroupRole { get; set; }

        public void Update(ExerciseMuscleGroup other)
        {
            MuscleGroupRole = other.MuscleGroupRole;
        }
    }
}

File: PeriodisationProgramApp.Web/src/types/ExerciseMuscleGroup.ts
import { MuscleGroupRole } from "../enums/MuscleGroupRole";
import { BaseEntity } from "./enitities/BaseEntity";
import { MuscleGroup } from "./enitities/MuscleGroup";

export type ExerciseMuscleGroup = {
  muscleGroup: MuscleGroup;
  muscleGroupRole: MuscleGroupRole;
} & BaseEntity;
